assets/src/blocks/Spreadsheet/SpreadsheetFrontend.js

```diff
@@ -16,88 +16,75 @@ const placeholderData = {
   ],
 };
 
-export class SpreadsheetFrontend extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      loading: null,
-      spreadSheetData: null,
-      searchText: '',
-      sortDirection: 'asc',
-      sortColumnIndex: null,
-    };
-
-    this.onHeaderClick = this.onHeaderClick.bind(this);
-  }
-
-  onHeaderClick(index) {
-    if (index === this.state.sortColumnIndex) {
-      const newDirection = this.state.sortDirection === 'asc' ? 'desc' : 'asc';
-      this.setState({
-        sortDirection: newDirection,
-      });
+export const SpreadsheetFrontend = ({
+  url,
+  color,
+  setInvalidSheetId,
+  className,
+}) => {
+  const [loading, setLoading] = useState(false);
+  const [spreadsheetData, setSpreadsheetData] = useState(null);
+  const [searchText, setSearchText] = useState('');
+  const [sortDirection, setSortDirection] = useState('asc');
+  const [sortColumnIndex, setSortColumnIndex] = useState(null);
+
+  const onHeaderClick = index => {
+    if (index === sortColumnIndex) {
+      const newDirection = sortDirection === 'asc' ? 'desc' : 'asc';
+      setSortDirection(newDirection);
```

Comment:
I think it's not necessary to define a `const` since it's only used once.

Maybe `setSortDirection(sortDirection === 'asc' ? 'desc' : 'asc');` is better?